File: panaderia-frontend/src/main/panaderia-frontend/src/app/modules/intranet/services/producto.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { ProductoBuscarRequest } from '../dto/request/producto-buscar.request';
import { ProductoRequest } from '../dto/request/producto.request';
import { FileResponse } from '../dto/response/file.response';
import { OutResponse } from '../dto/response/out.response';
import { ProductoResponse } from '../dto/response/producto.response';

@Injectable()
export class ProductoService {
  constructor(private http: HttpClient) { }

  public listarProducto(req: ProductoBuscarRequest): Observable<OutResponse<ProductoResponse[]>> {
    return this.http.post<OutResponse<ProductoResponse[]>>(`${environment.WsPanaderiaBackend}/producto/listarProducto`, req);
  }

  public registrarProducto(req: ProductoRequest): Observable<OutResponse<ProductoResponse>> {
    return this.http.post<OutResponse<ProductoResponse>>(`${environment.WsPanaderiaBackend}/producto/registrarProducto`, req);
  }

  public modificarProducto(req: ProductoRequest): Observable<OutResponse<ProductoResponse>> {
    return this.http.post<OutResponse<ProductoResponse>>(`${environment.WsPanaderiaBackend}/producto/modificarProducto`, req);
  }

  public eliminarProducto(req: ProductoRequest): Observable<OutResponse<ProductoResponse>> {
    return this.http.post<OutResponse<ProductoResponse>>(`${environment.WsPanaderiaBackend}/producto/eliminarProducto`, req);
  }

  public reporteXlsxListarProducto(req: ProductoBuscarRequest): Observable<OutResponse<FileResponse>> {
    return this.http.post<OutResponse<FileResponse>>(`${environment.WsPanaderiaBackend}/producto/reporteXlsxListarProducto`, req);
  }
}

File: panaderia-backend/src/main/java/com/besoft/panaderia/service/ProductoService.java
package com.besoft.panaderia.service;

import java.util.List;

import com.besoft.panaderia.dto.request.ProductoBuscarRequest;
import com.besoft.panaderia.dto.request.ProductoRequest;
import com.besoft.panaderia.dto.response.OutResponse;
import com.besoft.panaderia.dto.response.ProductoResponse;

public interface ProductoService {

	public OutResponse<List<ProductoResponse>> listarProducto(ProductoBuscarRequest req);

	public OutResponse<ProductoResponse> registrarProducto(ProductoRequest req);

	public OutResponse<ProductoResponse> modificarProducto(ProductoRequest req);

	public OutResponse<ProductoResponse> eliminarProducto(ProductoRequest req);
}
